### PR TITLE
documentation for writing payment modules

### PR DESCRIPTION
**Pull request by zcwilt**

### CODE REVIEW

File: content/dev/payment/writing_payment_modules/configuration_keys.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+   Whether that module is enabled or not
+
+ - MODULE_PAYMENT_%%_SORT_ORDER
+
+   The order that the module appears in both the admin and the catalog
+ - 
+ - MODULE_PAYMENT_%%_ZONE
+
+   Payment modules can be rerstricted to a certain geographic zone
```
Comment (drbyte) [inline content/dev/payment/writing_payment_modules/configuration_keys.md:26]:
```suggestion
   Payment modules can be restricted to a certain geographic zone
```

File: content/dev/payment/writing_payment_modules/configuration_keys.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## Custom keys
+
+Most payment modules will need to create their own custom keys/settings. 
+
+to do this the module must add a protected method to its class. 
+
+e.g. 
+
+```angular2html
```
Comment (drbyte) [inline content/dev/payment/writing_payment_modules/configuration_keys.md:40]:
```suggestion
```php
```

Comment (zcwilt) [reply content/dev/payment/writing_payment_modules/configuration_keys.md:40]: oops. for some reason phpstrm adds the angualar2html by default

File: content/dev/payment/writing_payment_modules/_index.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+This is required and should generally be the same as the class name for your payment module.
+
+The interface for the payment module also requires you to create a method in your class called 
+
+`addCustomConfigurationKeys`
+
+This method creates custom configuration keys for your payment module on top of the default configureation keys that are needed.
+
+See [Payment Module Configuration Keys]()
```
Comment (scottcwilson) [inline content/dev/payment/writing_payment_modules/_index.md:59]: See [Payment Module Configuration Keys](/dev/payment/writing_payment_modules/configuration_keys/)

Comment (scottcwilson) [reply content/dev/payment/writing_payment_modules/_index.md:59]: (Format of Hugo link is missing the second part.)

File: content/dev/payment/writing_payment_modules/_index.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+`public string $code = 'moneyorder';`   
+
+This is required and should generally be the same as the class name for your payment module.
+
+The interface for the payment module also requires you to create a method in your class called 
+
+`addCustomConfigurationKeys`
+
+This method creates custom configuration keys for your payment module on top of the default configureation keys that are needed.
```
Comment (drbyte) [inline content/dev/payment/writing_payment_modules/_index.md:57]:
```suggestion
This method creates custom configuration keys for your payment module on top of the default configuration keys that are needed.
```

**Comment by scottcwilson** [comment]
Ian, I will leave you to merge this when you're ready.

**Pull request closed without merging**